### PR TITLE
 Normalize configuration #12

### PR DESCRIPTION
**Pull request by ganchclub**

### CODE REVIEW

File: src/RecursivePagination.php [hunk 1]
```diff
@@ -34,6 +34,11 @@ class RecursivePagination {
      */
     public function __construct(Parser $parser, $xpath = null) {
       $this->parser = $parser;
+
+      if (!is_string($xpath) and !is_array($xpath)) {
+        throw new \InvalidArgumentException('Xpath should be an array or a string');
+      }
+
       if (isset($xpath)) {
```
Comment (funivan) [inline src/RecursivePagination.php]: remove this check

File: src/RecursivePagination.php [hunk 2]
```diff
@@ -34,6 +34,11 @@ class RecursivePagination {
      */
     public function __construct(Parser $parser, $xpath = null) {
```
Comment (funivan) [inline src/RecursivePagination.php]: required

File: src/RecursivePagination.php [hunk 3]
```diff
@@ -34,6 +34,11 @@ class RecursivePagination {
...
       $this->parser = $parser;
+
+      if (!is_string($xpath) and !is_array($xpath)) {
```
Comment (funivan) [inline src/RecursivePagination.php]: check for emtpy

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,13 @@
...
 All Notable changes to `RecursivePagination` will be documented in this file
 
-## Unreleased
+## Unreleased 0.1.0-alpha.1
+
+### Added
+- Xpath check added in constructor whether it's a string or an array 
+
+### Deprecated
+- remove parameter from `RecursivePagination::getNextPage()` method
```
Comment (funivan) [inline CHANGELOG.md]: I prefer `Remove first parameter from ...`

File: src/RecursivePagination.php [hunk 5]
```diff
@@ -92,17 +92,19 @@ public function getNextPage($customXpath = null) {
...
 
     /**
-     * @param string|array $xpath
+     * @param string|array $expression
      * @throws \InvalidArgumentException
      */
-    private function addXpath($xpath) {
-      if (!is_string($xpath) and !is_array($xpath)) {
-        throw new \InvalidArgumentException('Xpath should be an array or a string');
+    private function addExpression($expression) {
```
Comment (funivan) [inline src/RecursivePagination.php]: setExpression

File: src/RecursivePagination.php [hunk 6]
```diff
@@ -92,17 +92,19 @@ public function getNextPage($customXpath = null) {
...
+
+      if (!is_string($expression) and !is_array($expression) or empty($expression)) {
```
Comment (funivan) [inline src/RecursivePagination.php:100]: in this case just check type

File: src/RecursivePagination.php [hunk 7]
```diff
@@ -92,17 +92,19 @@ public function getNextPage($customXpath = null) {
...
+        throw new \InvalidArgumentException('Expression should be not empty array or string');
       }
-      $xpath = (array) $xpath;
-      foreach ($xpath as $path) {
+
+      $expression = (array) $expression;
+      foreach ($expression as $path) {
         if (!is_string($path)) {
```
Comment (funivan) [inline src/RecursivePagination.php]: check if we add not empty string

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,10 +4,10 @@ All Notable changes to `RecursivePagination` will be documented in this file
 ## Unreleased 0.1.0-alpha.1
 
 ### Added
-- Xpath check added in constructor whether it's a string or an array 
+- Expression check added in constructor whether it's a string or an array
```
Comment (funivan) [inline CHANGELOG.md:7]: +1

File: src/RecursivePagination.php (hunk 6 above)
Comment (ganchclub) [inline src/RecursivePagination.php:100]: in case with empty array this will not work

**Pull request merged**